enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,417 @@
+
+* Do we allow override of same named files in the old `extraManifestPath` field? 
+
+  * No, we do not. The old behavior does not change when customer define old `extraManifestPath` path and not the new field `extraManifestSearchPaths`.
+
+* What if customer include both `extraManifestPath` and `extraManifestSearchPaths` in the siteconfig?
+
+  * The behavior of `extraManifestSearchPaths` will take precedence. We will neither generate any CR from `extraManifestPath` nor the builtin CRs from the ztp-container
+
+
+
+### How does the proposal help
+
+
+* **Controllability:** The proposal provides the flexibility for the customer to include/exclude any CRs from multiple directories from user Git repository. 
+
+  * customer can put extra-manifests extracted from ztp-container at `myPath` and their own custom CRs in `myPath2` as shown below as an example 
+
+
+    ```yaml
+      - cluster:
+        extraManifestSearchPaths:  <------ new field
+          - myPath/
+          - myPath2/
+        extraManifests:
+          filter:			
+      ```
+
+  * Any same named CR file in `myPath2` will override from `myPath`. Prior to this change, the customer couldn't put same named CR in the extraManifestPath. Now overriding is possible but they have to opt in for `extraManifestSearchPaths`. 
+
+
+
+* **Version Independence**: 
+  * What is version independence in this context?
+    
+    * The ability to exclude the built in content and the user can set their system up to only ever pull from git. This means they can have a set of 4.12, 4.13, 4.14, ... extra manifests in git and point to the correct set based on what they are deploying (one tool supporting multiple versions).
+
+    * The decoupling of extra-manifests from ztp-container: current ztp-container only contain a single version of extra-manifests. Using the git approach, user doesn't need to launch different versioned ztp-container to deploy different versioned manifests, rather just point towards the correct git repo path.
+
+
+  This change will allow customer to have version independence of extra-manifests having different versioned folder in their git repository, as an example:
+
+
+```yaml
+siteconfig/
+├── kustomization.yaml
+├── version_4.12
+│   ├── kustomization.yaml
+│   ├── sno-extra-manifest/
+|   ├── custom-manifests/
+│   ├── cnfde9.yaml
+└── version_4.13
+    ├── kustomization.yaml
+    ├── sno-extra-manifest/
+    ├── custom-manifests/
+    ├── cnfde10.yaml    
+```
+
+### Workflow Description
+
+Variable scenarios are briefly explained in the below table based on the proposal and the output list of CRs:
+
+* Assumption: customers siteconfig consists 2 extra manifests paths which are in the user Git repository
+
+  ```yaml
+    - cluster:
+      extraManifestSearchPaths: 
+        - sno-extra-manifest/
+        - custom-manifests/
+      extraManifests:
```

Comment:
Can we do `.extraManifests.searchPaths` instead? Basically reading seachPath, filter etc are ops related to extraManifests ([see](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfig.go#L157-L160))

Reply:
Currently `extraManifestPath` (https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfig.go#L181) is outside of `extraManifests`